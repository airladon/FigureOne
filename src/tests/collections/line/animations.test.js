import { staticTester } from '../../staticTester';

const { getShapes, updates, move } = require('./animations.js');

staticTester('Collection: Line - Animations', getShapes, updates, move);
