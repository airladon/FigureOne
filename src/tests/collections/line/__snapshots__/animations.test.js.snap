// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection: Line - Animations Shape grow 1`] = `
Object {
  "border": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
  "elements": Object {
    "line": Object {
      "border": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
      "points": Array [
        0,
        0.025,
        0.4,
        0.025,
        0,
        -0.025,
        0,
        -0.025,
        0.4,
        0.025,
        0.4,
        -0.025,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
    },
  },
  "touchBorder": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
}
`;

exports[`Collection: Line - Animations Shape lengthAnimation 1`] = `
Object {
  "border": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
  "elements": Object {
    "line": Object {
      "border": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
      "points": Array [
        0,
        0.025,
        0.4,
        0.025,
        0,
        -0.025,
        0,
        -0.025,
        0.4,
        0.025,
        0.4,
        -0.025,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
    },
  },
  "touchBorder": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
}
`;

exports[`Collection: Line - Animations Shape lengthAnimationStep 1`] = `
Object {
  "border": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
  "elements": Object {
    "line": Object {
      "border": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
      "points": Array [
        0,
        0.025,
        0.4,
        0.025,
        0,
        -0.025,
        0,
        -0.025,
        0.4,
        0.025,
        0.4,
        -0.025,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
    },
  },
  "touchBorder": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
}
`;

exports[`Collection: Line - Animations Shape pulse-defaults 1`] = `
Object {
  "border": Array [
    Array [
      Array [
        -0.15,
        -0.005,
      ],
      Array [
        0.15,
        -0.005,
      ],
      Array [
        0.15,
        0.005,
      ],
      Array [
        -0.15,
        0.005,
      ],
    ],
    Array [
      Array [
        -0.15,
        0,
      ],
      Array [
        -0.14,
        0.03,
      ],
      Array [
        -0.2,
        0,
      ],
      Array [
        -0.14,
        -0.03,
      ],
    ],
    Array [
      Array [
        0.15,
        0,
      ],
      Array [
        0.14,
        -0.03,
      ],
      Array [
        0.2,
        0,
      ],
      Array [
        0.14,
        0.03,
      ],
    ],
    Array [
      Array [
        -0.035,
        0.065,
      ],
      Array [
        0.035,
        0.065,
      ],
      Array [
        0.035,
        0.137,
      ],
      Array [
        -0.035,
        0.137,
      ],
    ],
  ],
  "elements": Object {
    "arrow1": Object {
      "border": Array [
        Array [
          Array [
            -0.15,
            0,
          ],
          Array [
            -0.14,
            0.03,
          ],
          Array [
            -0.2,
            0,
          ],
          Array [
            -0.14,
            -0.03,
          ],
        ],
      ],
      "points": Array [
        0.05,
        0,
        0.06,
        0.03,
        0,
        0,
        0.05,
        0,
        0,
        0,
        0.06,
        -0.03,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.15,
            0,
          ],
          Array [
            -0.14,
            0.03,
          ],
          Array [
            -0.2,
            0,
          ],
          Array [
            -0.14,
            -0.03,
          ],
        ],
      ],
    },
    "arrow2": Object {
      "border": Array [
        Array [
          Array [
            0.15,
            0,
          ],
          Array [
            0.14,
            -0.03,
          ],
          Array [
            0.2,
            0,
          ],
          Array [
            0.14,
            0.03,
          ],
        ],
      ],
      "points": Array [
        -0.05,
        0,
        -0.06,
        -0.03,
        0,
        0,
        -0.05,
        0,
        0,
        0,
        -0.06,
        0.03,
      ],
      "touchBorder": Array [
        Array [
          Array [
            0.15,
            0,
          ],
          Array [
            0.14,
            -0.03,
          ],
          Array [
            0.2,
            0,
          ],
          Array [
            0.14,
            0.03,
          ],
        ],
      ],
    },
    "label": Object {
      "border": Array [
        Array [
          Array [
            -0.035,
            0.065,
          ],
          Array [
            0.035,
            0.065,
          ],
          Array [
            0.035,
            0.137,
          ],
          Array [
            -0.035,
            0.137,
          ],
        ],
      ],
      "elements": Object {
        "base": Object {
          "border": Array [
            Array [
              Array [
                -0.035,
                0.065,
              ],
              Array [
                0.035,
                0.065,
              ],
              Array [
                0.035,
                0.137,
              ],
              Array [
                -0.035,
                0.137,
              ],
            ],
          ],
          "touchBorder": Array [
            Array [
              Array [
                -0.035,
                0.065,
              ],
              Array [
                0.035,
                0.065,
              ],
              Array [
                0.035,
                0.137,
              ],
              Array [
                -0.035,
                0.137,
              ],
            ],
          ],
        },
      },
      "touchBorder": Array [
        Array [
          Array [
            -0.035,
            0.065,
          ],
          Array [
            0.035,
            0.065,
          ],
          Array [
            0.035,
            0.137,
          ],
          Array [
            -0.035,
            0.137,
          ],
        ],
      ],
    },
    "line": Object {
      "border": Array [
        Array [
          Array [
            -0.15,
            -0.005,
          ],
          Array [
            0.15,
            -0.005,
          ],
          Array [
            0.15,
            0.005,
          ],
          Array [
            -0.15,
            0.005,
          ],
        ],
      ],
      "points": Array [
        0,
        0.005,
        0.3,
        0.005,
        0,
        -0.005,
        0,
        -0.005,
        0.3,
        0.005,
        0.3,
        -0.005,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.15,
            -0.005,
          ],
          Array [
            0.15,
            -0.005,
          ],
          Array [
            0.15,
            0.005,
          ],
          Array [
            -0.15,
            0.005,
          ],
        ],
      ],
    },
  },
  "touchBorder": Array [
    Array [
      Array [
        -0.15,
        -0.005,
      ],
      Array [
        0.15,
        -0.005,
      ],
      Array [
        0.15,
        0.005,
      ],
      Array [
        -0.15,
        0.005,
      ],
    ],
    Array [
      Array [
        -0.15,
        0,
      ],
      Array [
        -0.14,
        0.03,
      ],
      Array [
        -0.2,
        0,
      ],
      Array [
        -0.14,
        -0.03,
      ],
    ],
    Array [
      Array [
        0.15,
        0,
      ],
      Array [
        0.14,
        -0.03,
      ],
      Array [
        0.2,
        0,
      ],
      Array [
        0.14,
        0.03,
      ],
    ],
    Array [
      Array [
        -0.035,
        0.065,
      ],
      Array [
        0.035,
        0.065,
      ],
      Array [
        0.035,
        0.137,
      ],
      Array [
        -0.035,
        0.137,
      ],
    ],
  ],
}
`;

exports[`Collection: Line - Animations Shape pulseWidth 1`] = `
Object {
  "border": Array [
    Array [
      Array [
        -0.15,
        -0.005,
      ],
      Array [
        0.15,
        -0.005,
      ],
      Array [
        0.15,
        0.005,
      ],
      Array [
        -0.15,
        0.005,
      ],
    ],
    Array [
      Array [
        -0.15,
        0,
      ],
      Array [
        -0.14,
        0.03,
      ],
      Array [
        -0.2,
        0,
      ],
      Array [
        -0.14,
        -0.03,
      ],
    ],
    Array [
      Array [
        0.15,
        0,
      ],
      Array [
        0.14,
        -0.03,
      ],
      Array [
        0.2,
        0,
      ],
      Array [
        0.14,
        0.03,
      ],
    ],
    Array [
      Array [
        -0.035,
        0.065,
      ],
      Array [
        0.035,
        0.065,
      ],
      Array [
        0.035,
        0.137,
      ],
      Array [
        -0.035,
        0.137,
      ],
    ],
  ],
  "elements": Object {
    "arrow1": Object {
      "border": Array [
        Array [
          Array [
            -0.15,
            0,
          ],
          Array [
            -0.14,
            0.03,
          ],
          Array [
            -0.2,
            0,
          ],
          Array [
            -0.14,
            -0.03,
          ],
        ],
      ],
      "points": Array [
        0.05,
        0,
        0.06,
        0.03,
        0,
        0,
        0.05,
        0,
        0,
        0,
        0.06,
        -0.03,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.15,
            0,
          ],
          Array [
            -0.14,
            0.03,
          ],
          Array [
            -0.2,
            0,
          ],
          Array [
            -0.14,
            -0.03,
          ],
        ],
      ],
    },
    "arrow2": Object {
      "border": Array [
        Array [
          Array [
            0.15,
            0,
          ],
          Array [
            0.14,
            -0.03,
          ],
          Array [
            0.2,
            0,
          ],
          Array [
            0.14,
            0.03,
          ],
        ],
      ],
      "points": Array [
        -0.05,
        0,
        -0.06,
        -0.03,
        0,
        0,
        -0.05,
        0,
        0,
        0,
        -0.06,
        0.03,
      ],
      "touchBorder": Array [
        Array [
          Array [
            0.15,
            0,
          ],
          Array [
            0.14,
            -0.03,
          ],
          Array [
            0.2,
            0,
          ],
          Array [
            0.14,
            0.03,
          ],
        ],
      ],
    },
    "label": Object {
      "border": Array [
        Array [
          Array [
            -0.035,
            0.065,
          ],
          Array [
            0.035,
            0.065,
          ],
          Array [
            0.035,
            0.137,
          ],
          Array [
            -0.035,
            0.137,
          ],
        ],
      ],
      "elements": Object {
        "base": Object {
          "border": Array [
            Array [
              Array [
                -0.035,
                0.065,
              ],
              Array [
                0.035,
                0.065,
              ],
              Array [
                0.035,
                0.137,
              ],
              Array [
                -0.035,
                0.137,
              ],
            ],
          ],
          "touchBorder": Array [
            Array [
              Array [
                -0.035,
                0.065,
              ],
              Array [
                0.035,
                0.065,
              ],
              Array [
                0.035,
                0.137,
              ],
              Array [
                -0.035,
                0.137,
              ],
            ],
          ],
        },
      },
      "touchBorder": Array [
        Array [
          Array [
            -0.035,
            0.065,
          ],
          Array [
            0.035,
            0.065,
          ],
          Array [
            0.035,
            0.137,
          ],
          Array [
            -0.035,
            0.137,
          ],
        ],
      ],
    },
    "line": Object {
      "border": Array [
        Array [
          Array [
            -0.15,
            -0.005,
          ],
          Array [
            0.15,
            -0.005,
          ],
          Array [
            0.15,
            0.005,
          ],
          Array [
            -0.15,
            0.005,
          ],
        ],
      ],
      "points": Array [
        0,
        0.005,
        0.3,
        0.005,
        0,
        -0.005,
        0,
        -0.005,
        0.3,
        0.005,
        0.3,
        -0.005,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.15,
            -0.005,
          ],
          Array [
            0.15,
            -0.005,
          ],
          Array [
            0.15,
            0.005,
          ],
          Array [
            -0.15,
            0.005,
          ],
        ],
      ],
    },
  },
  "touchBorder": Array [
    Array [
      Array [
        -0.15,
        -0.005,
      ],
      Array [
        0.15,
        -0.005,
      ],
      Array [
        0.15,
        0.005,
      ],
      Array [
        -0.15,
        0.005,
      ],
    ],
    Array [
      Array [
        -0.15,
        0,
      ],
      Array [
        -0.14,
        0.03,
      ],
      Array [
        -0.2,
        0,
      ],
      Array [
        -0.14,
        -0.03,
      ],
    ],
    Array [
      Array [
        0.15,
        0,
      ],
      Array [
        0.14,
        -0.03,
      ],
      Array [
        0.2,
        0,
      ],
      Array [
        0.14,
        0.03,
      ],
    ],
    Array [
      Array [
        -0.035,
        0.065,
      ],
      Array [
        0.035,
        0.065,
      ],
      Array [
        0.035,
        0.137,
      ],
      Array [
        -0.035,
        0.137,
      ],
    ],
  ],
}
`;

exports[`Collection: Line - Animations grow 1`] = `
Object {
  "border": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
  "elements": Object {
    "line": Object {
      "border": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
      "points": Array [
        0,
        0.025,
        0.4,
        0.025,
        0,
        -0.025,
        0,
        -0.025,
        0.4,
        0.025,
        0.4,
        -0.025,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
    },
  },
  "touchBorder": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
}
`;

exports[`Collection: Line - Animations lengthAnimation 1`] = `
Object {
  "border": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
  "elements": Object {
    "line": Object {
      "border": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
      "points": Array [
        0,
        0.025,
        0.4,
        0.025,
        0,
        -0.025,
        0,
        -0.025,
        0.4,
        0.025,
        0.4,
        -0.025,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
    },
  },
  "touchBorder": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
}
`;

exports[`Collection: Line - Animations lengthAnimationStep 1`] = `
Object {
  "border": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
  "elements": Object {
    "line": Object {
      "border": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
      "points": Array [
        0,
        0.025,
        0.4,
        0.025,
        0,
        -0.025,
        0,
        -0.025,
        0.4,
        0.025,
        0.4,
        -0.025,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.2,
            -0.025,
          ],
          Array [
            0.2,
            -0.025,
          ],
          Array [
            0.2,
            0.025,
          ],
          Array [
            -0.2,
            0.025,
          ],
        ],
      ],
    },
  },
  "touchBorder": Array [
    Array [
      Array [
        -0.2,
        -0.025,
      ],
      Array [
        0.2,
        -0.025,
      ],
      Array [
        0.2,
        0.025,
      ],
      Array [
        -0.2,
        0.025,
      ],
    ],
  ],
}
`;

exports[`Collection: Line - Animations pulse-defaults 1`] = `
Object {
  "border": Array [
    Array [
      Array [
        -0.15,
        -0.005,
      ],
      Array [
        0.15,
        -0.005,
      ],
      Array [
        0.15,
        0.005,
      ],
      Array [
        -0.15,
        0.005,
      ],
    ],
    Array [
      Array [
        -0.15,
        0,
      ],
      Array [
        -0.14,
        0.03,
      ],
      Array [
        -0.2,
        0,
      ],
      Array [
        -0.14,
        -0.03,
      ],
    ],
    Array [
      Array [
        0.15,
        0,
      ],
      Array [
        0.14,
        -0.03,
      ],
      Array [
        0.2,
        0,
      ],
      Array [
        0.14,
        0.03,
      ],
    ],
    Array [
      Array [
        -0.035,
        0.065,
      ],
      Array [
        0.035,
        0.065,
      ],
      Array [
        0.035,
        0.137,
      ],
      Array [
        -0.035,
        0.137,
      ],
    ],
  ],
  "elements": Object {
    "arrow1": Object {
      "border": Array [
        Array [
          Array [
            -0.15,
            0,
          ],
          Array [
            -0.14,
            0.03,
          ],
          Array [
            -0.2,
            0,
          ],
          Array [
            -0.14,
            -0.03,
          ],
        ],
      ],
      "points": Array [
        0.05,
        0,
        0.06,
        0.03,
        0,
        0,
        0.05,
        0,
        0,
        0,
        0.06,
        -0.03,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.15,
            0,
          ],
          Array [
            -0.14,
            0.03,
          ],
          Array [
            -0.2,
            0,
          ],
          Array [
            -0.14,
            -0.03,
          ],
        ],
      ],
    },
    "arrow2": Object {
      "border": Array [
        Array [
          Array [
            0.15,
            0,
          ],
          Array [
            0.14,
            -0.03,
          ],
          Array [
            0.2,
            0,
          ],
          Array [
            0.14,
            0.03,
          ],
        ],
      ],
      "points": Array [
        -0.05,
        0,
        -0.06,
        -0.03,
        0,
        0,
        -0.05,
        0,
        0,
        0,
        -0.06,
        0.03,
      ],
      "touchBorder": Array [
        Array [
          Array [
            0.15,
            0,
          ],
          Array [
            0.14,
            -0.03,
          ],
          Array [
            0.2,
            0,
          ],
          Array [
            0.14,
            0.03,
          ],
        ],
      ],
    },
    "label": Object {
      "border": Array [
        Array [
          Array [
            -0.035,
            0.065,
          ],
          Array [
            0.035,
            0.065,
          ],
          Array [
            0.035,
            0.137,
          ],
          Array [
            -0.035,
            0.137,
          ],
        ],
      ],
      "elements": Object {
        "base": Object {
          "border": Array [
            Array [
              Array [
                -0.035,
                0.065,
              ],
              Array [
                0.035,
                0.065,
              ],
              Array [
                0.035,
                0.137,
              ],
              Array [
                -0.035,
                0.137,
              ],
            ],
          ],
          "touchBorder": Array [
            Array [
              Array [
                -0.035,
                0.065,
              ],
              Array [
                0.035,
                0.065,
              ],
              Array [
                0.035,
                0.137,
              ],
              Array [
                -0.035,
                0.137,
              ],
            ],
          ],
        },
      },
      "touchBorder": Array [
        Array [
          Array [
            -0.035,
            0.065,
          ],
          Array [
            0.035,
            0.065,
          ],
          Array [
            0.035,
            0.137,
          ],
          Array [
            -0.035,
            0.137,
          ],
        ],
      ],
    },
    "line": Object {
      "border": Array [
        Array [
          Array [
            -0.15,
            -0.005,
          ],
          Array [
            0.15,
            -0.005,
          ],
          Array [
            0.15,
            0.005,
          ],
          Array [
            -0.15,
            0.005,
          ],
        ],
      ],
      "points": Array [
        0,
        0.005,
        0.3,
        0.005,
        0,
        -0.005,
        0,
        -0.005,
        0.3,
        0.005,
        0.3,
        -0.005,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.15,
            -0.005,
          ],
          Array [
            0.15,
            -0.005,
          ],
          Array [
            0.15,
            0.005,
          ],
          Array [
            -0.15,
            0.005,
          ],
        ],
      ],
    },
  },
  "touchBorder": Array [
    Array [
      Array [
        -0.15,
        -0.005,
      ],
      Array [
        0.15,
        -0.005,
      ],
      Array [
        0.15,
        0.005,
      ],
      Array [
        -0.15,
        0.005,
      ],
    ],
    Array [
      Array [
        -0.15,
        0,
      ],
      Array [
        -0.14,
        0.03,
      ],
      Array [
        -0.2,
        0,
      ],
      Array [
        -0.14,
        -0.03,
      ],
    ],
    Array [
      Array [
        0.15,
        0,
      ],
      Array [
        0.14,
        -0.03,
      ],
      Array [
        0.2,
        0,
      ],
      Array [
        0.14,
        0.03,
      ],
    ],
    Array [
      Array [
        -0.035,
        0.065,
      ],
      Array [
        0.035,
        0.065,
      ],
      Array [
        0.035,
        0.137,
      ],
      Array [
        -0.035,
        0.137,
      ],
    ],
  ],
}
`;

exports[`Collection: Line - Animations pulseWidth 1`] = `
Object {
  "border": Array [
    Array [
      Array [
        -0.15,
        -0.005,
      ],
      Array [
        0.15,
        -0.005,
      ],
      Array [
        0.15,
        0.005,
      ],
      Array [
        -0.15,
        0.005,
      ],
    ],
    Array [
      Array [
        -0.15,
        0,
      ],
      Array [
        -0.14,
        0.03,
      ],
      Array [
        -0.2,
        0,
      ],
      Array [
        -0.14,
        -0.03,
      ],
    ],
    Array [
      Array [
        0.15,
        0,
      ],
      Array [
        0.14,
        -0.03,
      ],
      Array [
        0.2,
        0,
      ],
      Array [
        0.14,
        0.03,
      ],
    ],
    Array [
      Array [
        -0.035,
        0.065,
      ],
      Array [
        0.035,
        0.065,
      ],
      Array [
        0.035,
        0.137,
      ],
      Array [
        -0.035,
        0.137,
      ],
    ],
  ],
  "elements": Object {
    "arrow1": Object {
      "border": Array [
        Array [
          Array [
            -0.15,
            0,
          ],
          Array [
            -0.14,
            0.03,
          ],
          Array [
            -0.2,
            0,
          ],
          Array [
            -0.14,
            -0.03,
          ],
        ],
      ],
      "points": Array [
        0.05,
        0,
        0.06,
        0.03,
        0,
        0,
        0.05,
        0,
        0,
        0,
        0.06,
        -0.03,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.15,
            0,
          ],
          Array [
            -0.14,
            0.03,
          ],
          Array [
            -0.2,
            0,
          ],
          Array [
            -0.14,
            -0.03,
          ],
        ],
      ],
    },
    "arrow2": Object {
      "border": Array [
        Array [
          Array [
            0.15,
            0,
          ],
          Array [
            0.14,
            -0.03,
          ],
          Array [
            0.2,
            0,
          ],
          Array [
            0.14,
            0.03,
          ],
        ],
      ],
      "points": Array [
        -0.05,
        0,
        -0.06,
        -0.03,
        0,
        0,
        -0.05,
        0,
        0,
        0,
        -0.06,
        0.03,
      ],
      "touchBorder": Array [
        Array [
          Array [
            0.15,
            0,
          ],
          Array [
            0.14,
            -0.03,
          ],
          Array [
            0.2,
            0,
          ],
          Array [
            0.14,
            0.03,
          ],
        ],
      ],
    },
    "label": Object {
      "border": Array [
        Array [
          Array [
            -0.035,
            0.065,
          ],
          Array [
            0.035,
            0.065,
          ],
          Array [
            0.035,
            0.137,
          ],
          Array [
            -0.035,
            0.137,
          ],
        ],
      ],
      "elements": Object {
        "base": Object {
          "border": Array [
            Array [
              Array [
                -0.035,
                0.065,
              ],
              Array [
                0.035,
                0.065,
              ],
              Array [
                0.035,
                0.137,
              ],
              Array [
                -0.035,
                0.137,
              ],
            ],
          ],
          "touchBorder": Array [
            Array [
              Array [
                -0.035,
                0.065,
              ],
              Array [
                0.035,
                0.065,
              ],
              Array [
                0.035,
                0.137,
              ],
              Array [
                -0.035,
                0.137,
              ],
            ],
          ],
        },
      },
      "touchBorder": Array [
        Array [
          Array [
            -0.035,
            0.065,
          ],
          Array [
            0.035,
            0.065,
          ],
          Array [
            0.035,
            0.137,
          ],
          Array [
            -0.035,
            0.137,
          ],
        ],
      ],
    },
    "line": Object {
      "border": Array [
        Array [
          Array [
            -0.15,
            -0.005,
          ],
          Array [
            0.15,
            -0.005,
          ],
          Array [
            0.15,
            0.005,
          ],
          Array [
            -0.15,
            0.005,
          ],
        ],
      ],
      "points": Array [
        0,
        0.005,
        0.3,
        0.005,
        0,
        -0.005,
        0,
        -0.005,
        0.3,
        0.005,
        0.3,
        -0.005,
      ],
      "touchBorder": Array [
        Array [
          Array [
            -0.15,
            -0.005,
          ],
          Array [
            0.15,
            -0.005,
          ],
          Array [
            0.15,
            0.005,
          ],
          Array [
            -0.15,
            0.005,
          ],
        ],
      ],
    },
  },
  "touchBorder": Array [
    Array [
      Array [
        -0.15,
        -0.005,
      ],
      Array [
        0.15,
        -0.005,
      ],
      Array [
        0.15,
        0.005,
      ],
      Array [
        -0.15,
        0.005,
      ],
    ],
    Array [
      Array [
        -0.15,
        0,
      ],
      Array [
        -0.14,
        0.03,
      ],
      Array [
        -0.2,
        0,
      ],
      Array [
        -0.14,
        -0.03,
      ],
    ],
    Array [
      Array [
        0.15,
        0,
      ],
      Array [
        0.14,
        -0.03,
      ],
      Array [
        0.2,
        0,
      ],
      Array [
        0.14,
        0.03,
      ],
    ],
    Array [
      Array [
        -0.035,
        0.065,
      ],
      Array [
        0.035,
        0.065,
      ],
      Array [
        0.035,
        0.137,
      ],
      Array [
        -0.035,
        0.137,
      ],
    ],
  ],
}
`;
